activesupport/lib/active_support/messages/metadata.rb

```diff
@@ -64,7 +64,11 @@ def match?(purpose)
         end
 
         def fresh?
-          @expires_at.nil? || Time.now.utc < Time.iso8601(@expires_at)
+          @expires_at.nil? || Time.now.utc < Time.iso8601(expires_at_iso8601)
+        end
+
+        def expires_at_iso8601
+          ActiveSupport.parse_json_times ?  @expires_at.iso8601 : @expires_at
```

Comment:
Don’t think we should know about this config here. We’d do a String type check or a respond to iso8601.

Can we move the iso8601 parsing to the initializer?